stats/counter32.go

```diff
@@ -0,0 +1,36 @@
+package stats
+
+import (
+	"sync/atomic"
+	"time"
+)
+
+type Counter32 struct {
+	val uint32
+}
+
+func NewCounter32(name string) *Counter32 {
+	return registry.add(name, &Counter32{}).(*Counter32)
+}
+
+func (c *Counter32) SetUint32(val uint32) {
+	atomic.StoreUint32(&c.val, val)
+}
+
+func (c *Counter32) Inc() {
+	atomic.AddUint32(&c.val, 1)
+}
+
+func (c *Counter32) Add(val int) {
+	c.AddUint32(uint32(val))
+}
+
+func (c *Counter32) AddUint32(val uint32) {
+	atomic.AddUint32(&c.val, val)
```

Comment:
On the `Counter64` there's a `SetUint64()` method, just for consistency (and to give a way of resetting) wouldn't it make sense to have that here too?